src/main/java/net/fabricmc/loader/transformer/accessWidener/AccessWidener.java

```diff
@@ -0,0 +1,240 @@
+			//Comment handling
+			int commentPos = line.indexOf('#');
+
+			if (commentPos >= 0) {
+				line = line.substring(0, commentPos).trim();
+			}
+
+			if (line.isEmpty()) continue;
+
+			//Will be a common issue, make it clear.
+			if (line.contains(" ")) {
+				throw new RuntimeException("AccessWidener contains one or more space character, tabs are required on line: " + line);
+			}
+
+			String[] split = line.split("\t");
+
+			if (split.length != 3 && split.length != 5) {
+				throw new RuntimeException(String.format("Invalid line (%s)", line));
+			}
+
+			String access = split[0];
+
+			targets.add(split[2].replaceAll("/", "."));
+
+			switch (split[1]) {
+			case "class":
+				if (split.length != 3) {
+					throw new RuntimeException(String.format("Expected (<access>\tclass\t<className>) got (%s)", line));
+				}
+
+				classAccess.put(split[2], applyAccess(access, classAccess.getOrDefault(split[2], Access.DEFAULT)));
+				break;
+			case "field":
+				if (split.length != 5) {
+					throw new RuntimeException(String.format("Expected (<access>\tfield\t<className>\t<fieldName>\t<fieldDesc>) got (%s)", line));
+				}
+
+				addOrMerge(fieldAccess, new EntryTriple(split[2], split[3], split[4]), access);
+				break;
+			case "method":
+				if (split.length != 5) {
+					throw new RuntimeException(String.format("Expected (<access>\tmethod\t<className>\t<methodName>\t<methodDesc>) got (%s)", line));
+				}
+
+				addOrMerge(methodAccess, new EntryTriple(split[2], split[3], split[4]), access);
+				break;
+			default:
+				throw new UnsupportedOperationException("Unsupported type " + split[1]);
+			}
+		}
+
+		Set<String> parentClasses = new LinkedHashSet<>();
+
+		//Also transform all parent classes
+		for (String clazz : targets) {
+			while (clazz.contains("$")) {
+				clazz = clazz.substring(0, clazz.lastIndexOf("$"));
+				parentClasses.add(clazz);
+			}
+		}
+
+		classes.addAll(targets);
+		classes.addAll(parentClasses);
+	}
+
+	void addOrMerge(Map<EntryTriple, Access> map, EntryTriple entry, String access) {
+		if (entry == null || access == null) {
+			throw new RuntimeException("Input entry or access is null");
+		}
+
+		map.put(entry, applyAccess(access, map.getOrDefault(entry, Access.DEFAULT)));
+	}
+
+	private Access applyAccess(String input, Access access) {
+		switch (input.toLowerCase(Locale.ROOT)) {
+		case "public":
+			return access.makePublic();
+		case "protected":
+			return access.makeProtected();
+		case "stripfinal":
+			return access.stripFinal();
+		default:
+			throw new UnsupportedOperationException("Unknown access type:" + input);
+		}
+	}
+
+	public Access getClassAccess(String className) {
+		return classAccess.getOrDefault(className, Access.DEFAULT);
+	}
+
+	public Access getFieldAccess(EntryTriple entryTriple) {
+		return fieldAccess.getOrDefault(entryTriple, Access.DEFAULT);
+	}
+
+	public Access getMethodAccess(EntryTriple entryTriple) {
+		return methodAccess.getOrDefault(entryTriple, Access.DEFAULT);
```

Comment:
I suggest considering method inheritance here.

Reply:
Player suggested to use tiny remapper for this (need exposing).

Even then I think it would be best for loom to handle it, I dont know of an easy / quick way to do it here.

Reply:
Imo loom does not suffice, given if you raise visibility of a method from `protected` to `public` in `net.minecraft.block.Block`, other mods may still override that method and have it as `protected`, causing a runtime error.

Reply:
Widening superclass method accessibility does not cause a run time error.

Reply:
Well, when the mod that raised the method to `public` calls the `protected` method, the runtime illegal access error occurs.

Reply:
That would be a compile time error in the development environment, not a run time error. Calling an overriding protected method by virtual dispatch over a public superclass method does not in itself cause a run time error.

Reply:
Mind send class files that I can test? Too lazy to generate with asm

Reply:
[Here is a Gist.](https://gist.github.com/kvverti/30058040da2def4116b79c72efc03d34)

Place the files in the appropriate package structure, then compile and run using the following steps.
1. Set access of `Super#m` to `protected`
2. `$ javac p/Super.java q/Sub.java`
3. Set access of `Super#m` to `public`
4. `$ javac p/Super.java`
5. `$ javac access/AccessTest.java access/UseM.java`
6. `$ java access.AccessTest`

You should get the output
```
Subclass
Subclass
```

Reply:
Unfortunately GitHub will not allow me to upload a zip of the class files

Reply:
Should be fine as removing final doesn't affect subclasses and protected still works.